### PR TITLE
feat(API): Remove Drupalisms from the API

### PR DESCRIPTION
**Pull request by e0ipso**
Task: #23

* [x] Ready for review
* [x] Ready for merge

Removes the Drupalisms from the API so the routes and fields are self explanatory.

### Notes
If you checkout the code in your local, explore the Docson schemas to see the results.

### QA

inspect schema contenta json api 2017-05-29 22-39-56

### CODE REVIEW

File: config/sync/jsonapi_extras.jsonapi_resource_config.block--block.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+resourceFields:
+  id:
+    fieldName: id
+    publicName: internalId
+    enhancer:
+      id: ''
+    disabled: false
+  status:
+    fieldName: status
+    publicName: isPublished
```
Comment (dawehner) [inline config/sync/jsonapi_extras.jsonapi_resource_config.block--block.yml:18]: Reading this made me realise my problem with renaming fields ... You suddenly have a split language between teams. Maybe you want that, maybe you won't

File: config/sync/jsonapi_extras.jsonapi_resource_config.block--block.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  langcode:
+    disabled: true
+    fieldName: langcode
+    publicName: langcode
+    enhancer:
+      id: ''
+  default_langcode:
+    disabled: true
+    fieldName: default_langcode
+    publicName: default_langcode
```
Comment (dawehner) [inline config/sync/jsonapi_extras.jsonapi_resource_config.block--block.yml:43]: It's weird that this field is not camelcase

File: config/sync/jsonapi_extras.jsonapi_resource_config.comment--comment.yml [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    disabled: false
+  subject:
+    fieldName: subject
+    publicName: subject
+    enhancer:
+      id: ''
+    disabled: false
+  uid:
+    fieldName: uid
+    publicName: author
```
Comment (dawehner) [inline config/sync/jsonapi_extras.jsonapi_resource_config.comment--comment.yml]: In Drupal we use owner, what about using that name here?

File: config/sync/jsonapi_extras.jsonapi_resource_config.comment--comment.yml [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+    disabled: false
+  thread:
+    fieldName: thread
+    publicName: thread
+    enhancer:
+      id: ''
+    disabled: false
+  entity_type:
+    fieldName: entity_type
+    publicName: entityType
```
Comment (dawehner) [inline config/sync/jsonapi_extras.jsonapi_resource_config.comment--comment.yml]: What about using entitytypeid?

File: config/sync/jsonapi_extras.jsonapi_resource_config.block--block.yml (hunk 2 above)
Comment (e0ipso) [reply config/sync/jsonapi_extras.jsonapi_resource_config.block--block.yml:43]: This field is disabled.

File: config/sync/jsonapi_extras.jsonapi_resource_config.comment--comment.yml (hunk 4 above)
Comment (e0ipso) [reply config/sync/jsonapi_extras.jsonapi_resource_config.comment--comment.yml]: That's probably more appropriate.

File: config/sync/jsonapi_extras.jsonapi_resource_config.comment--comment.yml (hunk 3 above)
Comment (e0ipso) [reply config/sync/jsonapi_extras.jsonapi_resource_config.comment--comment.yml]: I think that it makes sense to have the fields represent the actual relationship. Ownership of a comment (or a piece of content) is not necessarily the same as authorship. If I pay you to write an article, I am the owner and you are the author.

That being said, I don't have any feelings about this. I'll make the change to owner.

File: config/sync/jsonapi_extras.jsonapi_resource_config.block--block.yml (hunk 1 above)
Comment (e0ipso) [reply config/sync/jsonapi_extras.jsonapi_resource_config.block--block.yml:18]: Booleans are a yes or no response to a question. Note the difference:

— What's the _status_ of this node?
— Yes.

vs

— _Is_ this node _published_?
— Yes.

Comment (e0ipso) [reply config/sync/jsonapi_extras.jsonapi_resource_config.block--block.yml:18]: I think its positive to sever the ties between the names in the back-end and the names in the front-end.  As a front-end developer I don't want to know or care about your back-end implementation.

More than that, we want to show people that they don't need to marry a naming pattern from Drupal.

File: config/sync/jsonapi_extras.jsonapi_resource_config.comment--comment.yml (hunk 4 above)
Comment (e0ipso) [reply config/sync/jsonapi_extras.jsonapi_resource_config.comment--comment.yml]: Done.

File: config/sync/jsonapi_extras.jsonapi_resource_config.comment--comment.yml (hunk 3 above)
Comment (e0ipso) [reply config/sync/jsonapi_extras.jsonapi_resource_config.comment--comment.yml]: Done.

**Comment by e0ipso** [comment]
@dawehner this is ready for re-review.

File: config/sync/jsonapi_extras.jsonapi_resource_config.block--block.yml (hunk 2 above)
Comment (dawehner) [reply config/sync/jsonapi_extras.jsonapi_resource_config.block--block.yml:43]: oh I see. I think I can't deal with negative logic

File: config/sync/jsonapi_extras.jsonapi_resource_config.block--block.yml (hunk 1 above)
Comment (dawehner) [reply config/sync/jsonapi_extras.jsonapi_resource_config.block--block.yml:18]: That's a fair point, but well I think think this makes it a little bit harder in the internal team communication, but meh

**Pull request merged**